### ISSUE TITLE
Valid CSRF Token Required - Login page

### ISSUE THREAD

**Issue by dr33ble** [osTicket/osTicket#3256]
Hello All ,

I have read around and attempted some fixes but still having issues logging into the agent login with an error , before I have even entered details is says in red writing "authentication require" , I login with admin details and it just shows an error in the top right. "Valid CSRF Token Required".
running version v1.10-RC3. Im a newb to github so be gentle pls xD

**Comment by cosmospham** [comment]
**Confirm this bug.**

I checked the session table and see that `user_id` usually set as zero. After tracing the code, I see that `$thisstaff` usually is `NULL`, so the `user_id` value will be set as zero with:

```
// include/class.ostsession.php:199
',user_id='.db_input($thisstaff?$thisstaff->getId():0).
```

I wonder why we set `user_id=0`? Or this is caused by `$thisstaff` missing value? In my opinion, it is no need to set `user_id` is zero.

I think sometime a request from browser is sent to server when the `user_id` is zero, then is check and see no user there, and throw a 403 response with `Valid CSRF Token Required` message.
